apps/prairielearn/src/lib/preact-html.ts

```diff
@@ -0,0 +1,22 @@
+import { render } from 'preact-render-to-string/jsx';
+
+import { type HtmlSafeString, unsafeHtml } from '@prairielearn/html';
+import type { VNode } from '@prairielearn/preact-cjs';
+
+import { DEV_MODE } from './config.js';
```

Comment:
This is not safe. We can't just import config on the client (where this function is intended to be used).

Reply:
```
const Foo = ({ name }: { name: string }) => <div>Hello, {name}!</div>;
const a = performance.now();

for (let i = 0; i < 1_000_000; i++) {
  renderPreactToHtmlForClientSide(<Foo name="Peter" />);
}

const b = performance.now();
console.log('It took ' + (b - a) + ' ms.');
```

With pretty: It took 735.61681 ms.
Without pretty: It took 577.6832420000001 ms.

Reply:
For posterity: Peter and I discussed maybe making this conditional based on `process.env.NODE_ENV === 'production'` and configuring `esbuild` to substitute in the appropriate value there. We can leave that for a potential future PR.